File: JavaScript_Courses/JavaScript_Basics/Preparation/03_Conditional_Statements_Advanced_Exercise/08_On_Time_for_the_Exam.js
function onTimeForTheExam(input) {
    let hourExam = Number(input[0]);
    let minuteExam = Number(input[1]);
    let hourArrival = Number(input[2]);
    let minuteArrival = Number(input[3]);

    let examTimeAsMinutes = hourExam * 60 + minuteExam;
    let arrivalTimeAsMinutes = hourArrival * 60 + minuteArrival;

    let diff = arrivalTimeAsMinutes - examTimeAsMinutes;
    if (diff > 0) {
        console.log('Late');
        if (diff < 60) {
            console.log(`${(diff)} minutes after the start`);
        } else {
            let diffHours = Math.floor(diff / 60);
            let diffMins = diff % 60;
            if (diffMins < 10) {
                diffMins = `0${diffMins}`;
            }
            console.log(`${diffHours}:${diffMins} hours after the start`);
        }
    } else if (diff >= - 30) {
        console.log('On time');
        if (diff != 0) {
            console.log(`${(-diff)} minutes before the start`);
        }
    } else {
        console.log('Early');
        if (diff > -60) {
            console.log(`${(-diff)} minutes before the start`);
        } else {
            let diffHours = Math.floor((-diff) / 60);
            let diffMins = (-diff) % 60;
            if (diffMins < 10) {
                diffMins = `0${diffMins}`;
            }
            console.log(`${diffHours}:${diffMins} hours before the start`);
        }
    }

}


// onTimeForTheExam(["9", "30", "9", "50"]);
// onTimeForTheExam(["11", "30", "10", "55"]);

File: Python_Courses/Python_Basics/Python_PB_2023_10/03_Conditional_Statements_Advanced_Exercise/08_On_Time_for_the_Exam.py
exam_hour = int(input())
exam_min = int(input())
arrival_hour = int(input())
arrival_min = int(input())

exam_time_as_min = exam_hour * 60 + exam_min
arrival_time_as_min = arrival_hour * 60 + arrival_min

diff = arrival_time_as_min - exam_time_as_min
if diff > 0:
    print("Late")
    if diff < 60:
        print(f"{diff} minutes after the start")
    else:
        hh = diff // 60
        mm = diff % 60
        print(f"{hh}:{mm :02d} hours after the start")
elif diff >= -30:  # diff [-30; 0]
    print("On time")
    if not diff == 0:  # eq. if diff != 0:
        print(f"{abs(diff)} minutes before the start")
else:
    print("Early")
    if diff > -60:  # (-60; -30)
        print(f"{abs(diff)} minutes before the start")
    else:  # (-inf; -60]
        hh = abs(diff) // 60
        mm = abs(diff) % 60
        print(f"{hh}:{mm :02d} hours before the start")

# comment
# pass
# my_variable = 5
